src/_cffi_src/openssl/dh.py

```diff
@@ -227,4 +229,17 @@
     return DH_check(dh, ret);
 }
 #endif
+
+/* These functions were added in OpenSSL 1.1.0f commit d0c50e80a8 */
+#if CRYPTOGRAPHY_OPENSSL_LESS_THAN_110PRE6
```

Comment:
As your comment notes, the commit in question is not in a released version of 1.1.0 (1.1.0f is not yet released). This check says that anything newer than 1.1.0-pre6 has d2i_DHxparams_bio, which is incorrect.

Rather than trying to conditionally bind this based on the addition of macros in an unreleased version of OpenSSL I'd suggest just defining Cryptography_d2i_DHxparams_bio (and i2d) and just update the comment to note that we define our own to simplify support across all versions.